File: src/main/java/ukalus/blast/LevelScanner.java
package ukalus.blast;

import ukalus.level.Level;
import ukalus.math.Vector2D;
import java.util.function.Function;

public class LevelScanner implements Function<Vector2D, Boolean> {

  private static final long serialVersionUID = 3833468426717835826L;
  private Level level;

  public LevelScanner(Level level) {
    this.level = level;
  }

  public Boolean apply(Vector2D location) {
    return !level.contains(location) || level.get(location) == null;
  }

}

File: src/main/java/ukalus/blast/LevelScanner.kt
package ukalus.blast

import ukalus.level.Level
import ukalus.math.Vector2D
import java.util.function.Function

class LevelScanner(private val level: Level<*>) : Function<Vector2D, Boolean> {

    override fun apply(location: Vector2D) = !level.contains(location) || level[location] == null

}
